### ISSUE TITLE
Allow 'no-lines' around comment before else/else-if for `lines-around-comment`

### ISSUE THREAD

**Issue by blake-regalia** [eslint/eslint#10015]
**What rule do you want to change?**
`lines-around-comment`

**Does this change cause the rule to produce more or fewer warnings?**
Fewer.

**How will the change be implemented? (New option, new default behavior, etc.)?**
New object option: `"allowBeforeElse": true | false`

**Please provide some example code that this change will affect:**

```js
/*eslint lines-around-comment: ["warn", { "beforeLineComment": true }]*/

// combine bar and baz
let foo = bar + baz;

// input is a string
if('string' === typeof input) {
    return input+'!';
}
// input is not a string
else {
    throw new TypeError('input not a string');
}
```

**What does the rule currently do for this code?**
Warns `Expected line before comment` on `// input is not a string`

**What will the rule do after it's changed?**
Allow comments before else/else-if with the `"allowBeforeElse": true` option when `"beforeLineComment"` or `"beforeBlockComment"` is set to true.

**Justification**
No line breaks between the `if` and `else` block makes it easy to visually interpret that the two blocks belong to the same group.

**Comment by not-an-aardvark** [comment]
Thanks for the proposal. Personally, I'm not sure this style is common enough to be worth adding to the core `lines-around-comment` rule -- it seems like it would be a better fit for a custom rule. However, I'm not opposed to adding it if there are other team members that would be in favor of it.

**Comment by blake-regalia** [comment]
This has got to be more common than [`"allowArrayEnd"`](https://eslint.org/docs/rules/lines-around-comment#allowarrayend).

**Comment by not-an-aardvark** [comment]
It may be, but in my opinion we shouldn't lower standards for adding options to a rule just because it already has some questionable options. (`allowArrayEnd` was added in August 2015, and in the last few years we've generally become much more strict about what we add to rules, since we have a lot of rules and can't realistically support every user's preferences.)

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
